### PR TITLE
Added a tutorial for bundle adjustment in isis

### LINKED ISSUE #11: jigsaw tutorial
We need some kind of basic jigsaw tutorial. Autocnet is internal only, so I don't think we can create an Autocnet version of generating control networks. This would be an all-ISIS tutorial that would have to cover the process of: 

1. spiceinit a list of images 
2. footprintinit a list of images
3. Create the image adjacency file and creating a set of control points 
4. feature matching 
5. bundle adjustment.

### LINKED ISSUE #51: Document USGSCSM parameters
USGSCSM has parameters that are mostly used for configuring bundle adjustment: https://github.com/DOI-USGS/usgscsm/blob/main/src/UsgsAstroLsSensorModel.cpp#L52C6-L67

Having a breakdown somewhere on these parameters would be good. This is kind of a cross post, I think adding some basic inline comments would be good, but then adding these terms to the glossary here and maybe an explanation in our bundle docs would cover all of our bases.

### PR DESCRIPTION
**Pull request by AustinSanders**
Adds a bundle adjustment tutorial.  Should probably be fact checked, since I'm definitely not an expert in the art of bundle adjustment.

Closes #11 and Closes #51 

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by amystamile-usgs** [comment]
Might be worth including this information in here: 
[Jigsaw.pdf](https://github.com/DOI-USGS/asc-public-docs/files/15191314/Jigsaw.pdf)

**Comment by AustinSanders** [comment]
> Might be worth including this information in here: [Jigsaw.pdf](https://github.com/DOI-USGS/asc-public-docs/files/15191314/Jigsaw.pdf)

I'm struggling to integrate that information into a 'how-to-guide' -- a lot of it falls into application documentation and concepts.  Would it be acceptable to make a bundle adjustment concept doc that includes that info instead?

**Comment by amystamile-usgs** [comment]
> > Might be worth including this information in here: [Jigsaw.pdf](https://github.com/DOI-USGS/asc-public-docs/files/15191314/Jigsaw.pdf)
> 
> I'm struggling to integrate that information into a 'how-to-guide' -- a lot of it falls into application documentation and concepts. Would it be acceptable to make a bundle adjustment concept doc that includes that info instead?

Honestly this may be better on the actual jigsaw app docs page.

File: docs/how-to-guides/image-processing/bundle-adjustment-in-isis.md [hunk 1]
```diff
@@ -73,19 +89,51 @@ An explanation of the configuration options passed to autoseed is as follows:
...
 jigsaw fromlist=image_list.lis cnet=control_network.net onet=control_network_jig.net update=no file_prefix=jig maxits=10 camsolve=angles camera_angles_sigma=.25
 ```
 
-An explanation of the configuration options passed to jigsaw is as follows:
+#### Option 2: For images with a CSM State String
+If you chose option 2 (csminit) for step 2, then you must also choose this option for jigsaw. When using jigsaw to bundle adjust a `csminit`-ed image, the user must pass CSM-specific parameter names via the `csmsolvelist` parameter.  Allowable values and their descriptions are as follows.
+
+| Parameter | Description |
+|---|---|
+| IT Pos. Bias     |  "In Track Position Bias" - a constant shift in the spacecraft's position parallel to the flight path |
```
Comment (jlaura) [inline docs/how-to-guides/image-processing/bundle-adjustment-in-isis.md:108]: It looks like units are needed for all of these. Also, consider telling folks where are correlated. For example, solving for CT Pos. Bias and Phi.

File: docs/how-to-guides/image-processing/bundle-adjustment-in-isis.md [hunk 2]
```diff
@@ -11,16 +11,32 @@ Bundle adjustment inherently requires multiple observations, and ISIS expects th
 ls -d -1 "$PWD/"*.cub > image_list.lis
```
Comment (jlaura) [inline docs/how-to-guides/image-processing/bundle-adjustment-in-isis.md:11]: Am I missing the part where the person downloads the images and control network to run through this tutorial?

File: docs/how-to-guides/image-processing/bundle-adjustment-in-isis.md [hunk 3]
```diff
@@ -11,16 +11,32 @@ Bundle adjustment inherently requires multiple observations, and ISIS expects th
...
+
+This tutorial assumes that ISDs are named by simply appending a .json extension to the full path of the .cub, i.e. the ISD for `my_cube.cub` is stored in `my_cube.cub.json`.  If this requirement is satisfied, then the user can quickly and easily `csminit` a series of images with the batchlist parameter as follows:
+
+```console
+csminit -batchlist=image_list.lis from=\$1 isd=\$1.json
+```
+
 ## Step 3: Finding the Geospatial Extents of Images with footprintinit
 
 The geospatial extent of the image refers to the lat/lon bounding box of the image, i.e. the image's "footprint." The lat/lon polygon of the image's extent can be attached to an image using ISIS's `footprintinit` program.  Similarly to the process of `spiceinit`ing a list of images, `footprintinit`ing a list of images is best used in conjunction with the `batchlist` parameter as follows:
```
Comment (jlaura) [inline docs/how-to-guides/image-processing/bundle-adjustment-in-isis.md:42]: Missing CLI invocation.

**Review by jlaura — COMMENTED** [review]
This reads more like a high level, general about than a tutorial. If that is the intention 👍. If not, I believe this needs to have a small amount of actual data with described inputs and outputs that one can copy/paste to run. 

Then is needs a discussion of the outputs. Right now, someone is going to get a wall of text/numbers and have no idea what they are looking at.

File: docs/how-to-guides/image-processing/bundle-adjustment-in-isis.md (hunk 3 above)
Comment (AustinSanders) [reply docs/how-to-guides/image-processing/bundle-adjustment-in-isis.md:42]: The footprintinit CLI invocation is on line 45

File: docs/how-to-guides/image-processing/bundle-adjustment-in-isis.md (hunk 1 above)
Comment (AustinSanders) [reply docs/how-to-guides/image-processing/bundle-adjustment-in-isis.md]: Good call on adding units, but I'm at a loss here. csminit.xml shows that all parameters are "m," and the code shows that getParameterUnits() is hard-coded to "m" with a note that "All parameters are meters or scaled to meters."  Does that seem correct...?

**Comment by jrcain-usgs** [comment]
As somewhat of a newbie, I was able to go through the tutorial and get the same result at the end.  (I only have a basic understanding of the whole bundle-adjustment process, may read more into jigsaw docs later as suggested at the top.)  The data added made it much easier to follow along.

**Review by jrcain-usgs — APPROVED** [review]

**Pull request merged**